### ISSUE TITLE
Question: OpenSearch on AWS cloud ?

### ISSUE THREAD

**Issue by alexey2baranov** [vduseev/opensearch-logger#12]
Hello! 
I hope  it is possible to to forward logs to AWS Cloud  OpenSearch ?  If so , could you provide additional section in documentation for such case please.

Thanks

currently I get warn 
/home/aleksei/.cache/pypoetry/virtualenvs/openai-assistant-tNPVYGJS-py3.10/lib/python3.10/site-packages/urllib3/connectionpool.py:1100: InsecureRequestWarning: Unverified HTTPS request is being made to host 'search-your-opensearch-domain-gudntas4y3f2ut4xbdl7opf4aq.us-east-1.es.amazonaws.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(

**Comment by vduseev** [comment]
Hi @alexey2baranov, yes it is most definitely possible. In fact, this library was primarily developed with managed AWS OpenSearch version in mind. 

To me it looks like some settings in your connection are incorrect. Could you please share the entire code chunk that you use to connect to the cloud OpenSearch instance?

**Comment by alexey2baranov** [comment]
I also find this source. How do you think is your plugin compatible with it ?
https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sources/http-source/

**Comment by vduseev** [comment]
I think you need to set verify_certs parameter to True.

As for the data prepper it's a completely separate server side pipeline solution that was recently made a part of OpenSearch suite. You even need to run it separately, because it is not included into all-in-one OpenSearch distribution.

I kind of fail to see how it is related in any way to what we are doing here.

If you are trying to send logs using this library to the data prepper pipeline using the http source, then no, I don't think it's going to work.

**Comment by vduseev** [comment]
You can also find an example of connecting to AWS OpenSearch in the Readme. It uses boto3 credentials object.

Instantiation of the boto3 object is outside of scope of this guide and I recommend checking out the official boto3 docs.

**Comment by alexey2baranov** [comment]
> I think you need to set verify_certs parameter to True.
> 
> As for the data prepper it's a completely separate server side pipeline solution that was recently made a part of OpenSearch suite. You even need to run it separately, because it is not included into all-in-one OpenSearch distribution.
> 
> I kind of fail to see how it is related in any way to what we are doing here.
> 
> If you are trying to send logs using this library to the data prepper pipeline using the http source, then no, I don't think it's going to work.

Thank you for clarification. Actualy it's my first time when I work with open search , so my question may seems of logic

**Comment by alexey2baranov** [comment]
Thank you for pointing to the AWS section into readme, I missed it. I'm well awared about boto3 as we heavely use AWS in the project. so I will try to reconfigure your plugin and write you later about results

**Comment by alexey2baranov** [comment]
well  I have progress. Now it pushes logs to AWS and this it great. Now I wondering why it stringify my json objct from logs? for the following test i got the following log, which is string "{'idd': 4, 'name': 'test 4'}"

```python
    for i in range(5):
        logger.info({"idd": i, "name": f"test {i}"})
        await asyncio.sleep(0.1)
```

```
{
  "_index": "demo-2024.02.09",
  "_type": "_doc",
  "_id": "2TwNkI0BRsHWoUyuJZjo",
  "_version": 1,
  "_score": null,
  "_source": {
    "ecs": {
      "version": "1.4.0"
    },
    "agent": {
      "ephemeral_id": "dfb13f3a-6067-4f19-9fde-74cdf9b01ebb",
      "type": "opensearch-logger",
      "version": "1.3.0"
    },
    "host": {
      "hostname": "EPRSBELL001B",
      "name": "EPRSBELL001B",
      "id": "EPRSBELL001B",
      "ip": "127.0.1.1"
    },
    "@timestamp": "2024-02-09T22:46:36.500Z",
    "message": "{'idd': 4, 'name': 'test 4'}",
    "log": {
      "original": "{'idd': 4, 'name': 'test 4'}",
      "level": "INFO",
      "logger": "divna.opensearch.test",
      "origin": {
        "file": {
          "line": 42,
          "name": "open_search_test.py",
          "path": "/home/aleksei/code/openai-assistant/tests/open_search_test.py"
        },
        "function": "test_log_to_opensearch",
        "module": "open_search_test"
      },
      "process": {
        "name": "MainProcess",
        "pid": 2643339
      },
      "thread": {
        "name": "MainThread",
        "id": 140219215135168
      }
    },
    "args": [],
    "stack_info": ""
  },
  "fields": {
    "@timestamp": [
      "2024-02-09T22:46:36.500Z"
    ]
  },
  "sort": [
    1707518796500
  ]
}
```

**Comment by vduseev** [comment]
I'm going to close this issue since the assistance with the original inquiry has been provided.

We are now in general Python questions territory and something like Stack Overflow would be a better place to ask those. I'd recommend to do a search on how to pass a dictionary to a logger handler in Python. This question is not related to this library specifically. The same would happen if we passed a dictionary to any other logger handler as a first argument.

Nevertheless, below is the answer to this question, for completeness. Any logger implemented according to the Python logger standard is supposed to stringify whatever is passed as the first argument. To pass the dictionary it up as extra fields, use the `extra` argument. The [official Python logging module documentation](https://docs.python.org/3/library/logging.html) is a great source of knowledge about working with logging in Python and expands on how all that works under the hood.

TL;DR: The correct way to pass a `dict` to a logger in Python is via the `extra` argument:

```python
for i in range(5):
        logger.info("my log line", extra={"idd": i, "name": f"test {i}"})
        await asyncio.sleep(0.1)
```

Hope that helps!

**Comment by alexey2baranov** [comment]
Onestly it's the best issue support i ever seen on github. Thank you very much for your effort @vduseev

**Comment by vduseev** [comment]
Thank you! Glad I could help!
